src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/UsageDetails/GetAzureRmConsumptionUsageDetail.cs

```diff
@@ -0,0 +1,159 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Consumption.Common;
+using Microsoft.Azure.Commands.Consumption.Models;
+using Microsoft.Azure.Management.Consumption;
+using Microsoft.Azure.Management.Consumption.Models;
+using Microsoft.Rest.Azure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Consumption.Cmdlets.UsageDetails
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmConsumptionUsageDetail", DefaultParameterSetName = Constants.ParameterSetNames.SubscriptionItemParameterSet), OutputType(typeof(List<PSUsageDetail>))]
+    public class GetAzureRmConsumptionUsageDetail : AzureConsumptionCmdletBase
+    {
+        const int MaxNumberToFetch = 1000;
```

Comment:
this is still needed. This is the number of items to fetch in each request. as suggested by Mark, we keep pulling the next page until we have everything. but API has output size limit, thus we pull max 1000 result in each request to API call.